libsql-server/src/schema/migration.rs

```diff
@@ -144,7 +147,20 @@ pub fn perform_migration<B: QueryResultBuilder>(
                 savepoint.commit().unwrap();
                 MigrationTaskStatus::Success
             };
-            (Ok(b), status)
+            (Ok(builder.into_inner()), status)
+        }
+        Ok(builder) => {
+            assert!(!builder.is_success());
+            let status = if dry_run {
+                savepoint.rollback().unwrap();
+                drop(savepoint);
+                MigrationTaskStatus::DryRunFailure
+            } else {
+                savepoint.commit().unwrap();
+                MigrationTaskStatus::Failure
+            };
```

Comment:
Wouldn't it be better to always rollback on failure?

Reply:
you're right